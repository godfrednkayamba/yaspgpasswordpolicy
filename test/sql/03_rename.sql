--suppress "MD5 password cleared because of role rename" messages
SET client_min_messages TO warning;

CREATE USER aaa PASSWORD 'DummY';
CREATE USER bbb;

LOAD 'yaspgpp';
--
--reset all settings
--
SET yaspgpp.username_min_length TO DEFAULT;
SET yaspgpp.username_min_special TO DEFAULT;
SET yaspgpp.username_min_upper TO DEFAULT;
SET yaspgpp.username_min_upper TO DEFAULT;
SET yaspgpp.username_min_digit TO DEFAULT;
SET yaspgpp.username_contain_password TO DEFAULT;
SET yaspgpp.username_ignore_case TO DEFAULT;
SET yaspgpp.username_contain TO DEFAULT;
SET yaspgpp.username_not_contain TO DEFAULT;
SET yaspgpp.username_min_repeat TO DEFAULT;
SET yaspgpp.password_min_length TO DEFAULT;
SET yaspgpp.password_min_special TO DEFAULT;
SET yaspgpp.password_min_upper TO DEFAULT;
SET yaspgpp.password_min_upper TO DEFAULT;
SET yaspgpp.password_min_digit TO DEFAULT;
SET yaspgpp.password_contain_username TO DEFAULT;
SET yaspgpp.password_ignore_case TO DEFAULT;
SET yaspgpp.password_contain TO DEFAULT;
SET yaspgpp.password_not_contain TO DEFAULT;
SET yaspgpp.password_min_repeat TO DEFAULT;

--
--length must be >=2
--
SET yaspgpp.username_min_length TO 2;
DROP USER IF EXISTS b;
ALTER USER aaa RENAME TO b;
-- Check that renaiming a user without password also invoke the extension
ALTER USER bbb RENAME TO b;
DROP USER bbb;
CREATE USER b;

--
--min user repeat
--
SET yaspgpp.username_min_repeat TO 5;
ALTER USER aaa RENAME TO abbbaaaaaa;
--
--min special >= 1
--
SET yaspgpp.username_min_special TO 1;
ALTER USER aaa RENAME TO bbb;
--
--min upper >=1
--
SET yaspgpp.username_min_upper TO 1;
ALTER USER aaa RENAME TO "b$bb";
--
--min lower >=2
--
SET yaspgpp.username_min_lower TO 1;
ALTER USER aaa RENAME TO "B$BB";
--
--must contain one of the characters 'a','b','c'
--
SET yaspgpp.username_contain TO 'a,b,c';
ALTER USER aaa RENAME TO "d$eF";
--
--must not contain one of the characters 'x','z'
--
SET yaspgpp.username_not_contain TO 'x,z';
ALTER USER aaa RENAME TO "a$exF";
--
--min digit >=1
--
SET yaspgpp.username_min_digit TO 1;
ALTER USER aaa RENAME TO "a$eFD";

DROP USER aaa;
