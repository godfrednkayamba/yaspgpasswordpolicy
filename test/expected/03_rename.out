--suppress "MD5 password cleared because of role rename" messages
SET client_min_messages TO warning;
CREATE USER aaa PASSWORD 'DummY';
CREATE USER bbb;
LOAD 'yaspgpasswordpolicy';
--
--reset all settings
--
SET yaspgpp.username_min_length TO DEFAULT;
SET yaspgpp.username_min_special TO DEFAULT;
SET yaspgpp.username_min_upper TO DEFAULT;
SET yaspgpp.username_min_upper TO DEFAULT;
SET yaspgpp.username_min_digit TO DEFAULT;
SET yaspgpp.username_contain_password TO DEFAULT;
SET yaspgpp.username_ignore_case TO DEFAULT;
SET yaspgpp.username_contain TO DEFAULT;
SET yaspgpp.username_not_contain TO DEFAULT;
SET yaspgpp.username_min_repeat TO DEFAULT;
SET yaspgpp.password_min_length TO DEFAULT;
SET yaspgpp.password_min_special TO DEFAULT;
SET yaspgpp.password_min_upper TO DEFAULT;
SET yaspgpp.password_min_upper TO DEFAULT;
SET yaspgpp.password_min_digit TO DEFAULT;
SET yaspgpp.password_contain_username TO DEFAULT;
SET yaspgpp.password_ignore_case TO DEFAULT;
SET yaspgpp.password_contain TO DEFAULT;
SET yaspgpp.password_not_contain TO DEFAULT;
SET yaspgpp.password_min_repeat TO DEFAULT;
--
--length must be >=2
--
SET yaspgpp.username_min_length TO 2;
ALTER USER aaa RENAME TO b;
ERROR:  username length should match the configured yaspgpp.username_min_length (2)
-- Check that renaiming a user without password also invoke the extension
ALTER USER bbb RENAME TO b;
ERROR:  username length should match the configured yaspgpp.username_min_length (2)
DROP USER bbb;
CREATE USER b;
ERROR:  username length should match the configured yaspgpp.username_min_length (2)
--
--min user repeat
--
SET yaspgpp.username_min_repeat TO 5;
ALTER USER aaa RENAME TO abbbaaaaaa;
ERROR:  username characters are repeated more than the configured yaspgpp.username_min_repeat times (5)
--
--min special >= 1
--
SET yaspgpp.username_min_special TO 1;
ALTER USER aaa RENAME TO bbb;
ERROR:  username does not contain the configured yaspgpp.username_min_special characters (1)
--
--min upper >=1
--
SET yaspgpp.username_min_upper TO 1;
ALTER USER aaa RENAME TO "b$bb";
ERROR:  username does not contain the configured yaspgpp.username_min_upper characters (1)
--
--min lower >=2
--
SET yaspgpp.username_min_lower TO 1;
ALTER USER aaa RENAME TO "B$BB";
ERROR:  username does not contain the configured yaspgpp.username_min_lower characters (1)
--
--must contain one of the characters 'a','b','c'
--
SET yaspgpp.username_contain TO 'a,b,c';
ALTER USER aaa RENAME TO "d$eF";
ERROR:  username does not contain the configured yaspgpp.username_contain characters: a,b,c
--
--must not contain one of the characters 'x','z'
--
SET yaspgpp.username_not_contain TO 'x,z';
ALTER USER aaa RENAME TO "a$exF";
ERROR:  username contains the configured yaspgpp.username_not_contain unauthorized characters: x,z
--
--min digit >=1
--
SET yaspgpp.username_min_digit TO 1;
ALTER USER aaa RENAME TO "a$eFD";
ERROR:  username does not contain the configured yaspgpp.username_min_digit characters (1)
DROP USER aaa;
